packages/plugin-ext/src/main/browser/menus/menus-contribution-handler.ts

```diff
@@ -266,36 +266,50 @@ export class MenusContributionPointHandler {
             const uri = this.resourceContextKey.get();
             return uri ? uri['codeUri'] : undefined;
         };
-        this.registerMenuAction(menuPath, menu, {
-            execute: () => this.commands.executeCommand(menu.command, selectedResource()),
-            isEnabled: () => this.commands.isEnabled(menu.command, selectedResource()),
-            isVisible: () => this.commands.isVisible(menu.command, selectedResource())
-        });
+        this.registerMenuAction(menuPath, menu, command => ({
+            execute: () => this.commands.executeCommand(command, selectedResource()),
+            isEnabled: () => this.commands.isEnabled(command, selectedResource()),
+            isVisible: () => this.commands.isVisible(command, selectedResource())
+        }));
     }
 
-    protected registerMenuAction(menuPath: MenuPath, menu: Menu, handler: CommandHandler): void {
-        const commandId = this.createSyntheticCommandId(menu, { prefix: '__plugin.menu.action.' });
+    protected registerMenuAction(menuPath: MenuPath, menu: Menu, handler: (command: string) => CommandHandler): void {
+        const commandId = this.createSyntheticCommandId(menu.command, { prefix: '__plugin.menu.action.' });
         const command: Command = { id: commandId };
-        this.commands.registerCommand(command, handler);
+        this.commands.registerCommand(command, handler(menu.command));
+
+        const altId = menu.alt && this.createSyntheticCommandId(menu.alt, { prefix: '__plugin.menu.action.' });
+        const alt: Command | undefined = altId && { id: altId } || undefined;
+        if (alt) {
+            this.commands.registerCommand(alt, handler(menu.alt!));
+        }
 
         const { when } = menu;
         const [group = '', order = undefined] = (menu.group || '').split('@');
-        const action: MenuAction = { commandId, order, when };
+        const action: MenuAction = { commandId, alt: altId, order, when };
         const inline = /^inline/.test(group);
         menuPath = inline ? menuPath : [...menuPath, group];
         this.menuRegistry.registerMenuAction(menuPath, action);
 
         this.onDidRegisterCommand(menu.command, pluginCommand => {
             command.category = pluginCommand.category;
-            action.label = pluginCommand.label;
+            command.label = pluginCommand.label;
             if (inline) {
-                action.icon = pluginCommand.iconClass;
+                command.iconClass = pluginCommand.iconClass;
```

Comment:
[question]: Why did you switch from `icon` to `iconClass`? Was this change related to the PR? I am fine going with your changes, just asking though.

Reply:
it supposed to be in the scope of PR. If an `action` has icon then it is used regardless `iconClass` of a command. But now then you ask i wonder whether it is really necessary and won't this synthetic command now appear in the quick palette. We did not want it to appear. I will check again and change as required.

Reply:
Well, it is certainly bogus :) 

<img width="621" alt="Screen Shot 2019-09-03 at 08 50 17" src="https://user-images.githubusercontent.com/3082655/64143057-e86cc680-ce27-11e9-9caa-bc8f295ae9a3.png">

Reply:
I've refactored to disable synthetic command in the command palette: https://github.com/theia-ide/theia/pull/6069/files#diff-c111d7d3381b06ddff0ddb201236bec1R287 and https://github.com/theia-ide/theia/pull/6069/files#diff-c111d7d3381b06ddff0ddb201236bec1R280